library/swift/src/Envoy.swift

```diff
@@ -42,3 +45,22 @@ public final class Envoy: NSObject {
     }
   }
 }
+
+extension Envoy: Client {
+  public func startStream(with request: Request, handler: ResponseHandler) -> StreamEmitter {
+    let httpStream = self.engine.startStream(with: handler.underlyingObserver)
+    httpStream.sendHeaders(request.outboundHeaders(), close: false)
+    return EnvoyStreamEmitter(stream: httpStream)
+  }
+
+  public func sendUnary(_ request: Request, body: Data?,
```

Comment:
Instead of `startStream` and `sendUnary`, I'd suggest:
`send(request:handler)` and `send(request:body:handler)`

Reply:
or `sendRequest(_:handler:)` / `sendRequest(_:body:handler:)`

Reply:
I'd rather lean towards being clear on the intent (stream/unary) for the end consumer's sake, rather than making them assume based on function parameter signatures

Reply:
And of course trailers can be in there too, but since trailers are so rare in normal cases, I'd probably still suggest a signature which excludes or defaults them.

Reply:
I understand the argument about clarity/specificity, but I doubt most end-users are likely to be thinking in these terms, and I have misgivings because I don't really think this actually more accurate.

The `startStream` interface is a perfectly appropriate interface for sending a "unary" request, and the current naming sort of makes it sound like you should only use it when you're trying to stream something (a naive user might even think that this is to stream a response, which is completely unrelated).

In fact, all HTTP requests are "unary" in the sense that they have a single request and response. I feel like these are semi-GRPC concepts that are leaking in because GRPC layers a stream encoding into a "unary" HTTP request.

Reply:
I guess in general, since this is a top level interface, I'm pushing back a little because I feel the naming should be recognizable and understandable to users who don't have a deep familiarity with the underlying concerns.

Reply:
Talked offline - this makes sense. Going to update in a separate PR after merging this

Reply:
https://github.com/lyft/envoy-mobile/pull/354